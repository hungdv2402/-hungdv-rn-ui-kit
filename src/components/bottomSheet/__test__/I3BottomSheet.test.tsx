import renderer from 'react-test-renderer';
import I3BottomSheet from '../I3BottomSheet';
import React from 'react';
import { Text } from 'react-native';

jest.mock('react-native-modal', () => 'Modal');

describe('renders I3BottomSheet correctly', () => {
  it('should render I3BottomSheet with basic properties', () => {
    const tree = renderer
      .create(
        <I3BottomSheet
          title='Modal Title'
          visible={true}
          closeModal={jest.fn()}
        >
          <Text>should render I3BottomSheet with basic properties</Text>
        </I3BottomSheet>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render I3BottomSheet with useSwipeDown={true} and title="Modal Title"', () => {
    const tree = renderer
      .create(
        <I3BottomSheet
          title='Modal Title'
          useSwipeDown
          visible={true}
          closeModal={jest.fn()}
        >
          <Text>should render I3BottomSheet with useSwipeDown={true}</Text>
        </I3BottomSheet>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});