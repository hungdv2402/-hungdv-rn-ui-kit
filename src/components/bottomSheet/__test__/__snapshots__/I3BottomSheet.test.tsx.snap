// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders I3BottomSheet correctly should render I3BottomSheet with basic properties 1`] = `
<Modal
  animationIn="slideInUp"
  animationOut="slideOutDown"
  hasBackdrop={true}
  isVisible={true}
  onBackdropPress={[MockFunction]}
  style={
    {
      "margin": 0,
    }
  }
  useNativeDriverForBackdrop={true}
>
  <View
    style={
      {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "marginTop": "auto",
        "padding": 16,
        "paddingBottom": 40,
      }
    }
  >
    <View
      style={
        {
          "height": 32,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Modal Title
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/components/bottomSheet/assets/close.png",
            }
          }
          style={
            {
              "height": 28,
              "width": 28,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "height": 16,
          "width": undefined,
        }
      }
    />
    <Text>
      should render I3BottomSheet with basic properties
    </Text>
  </View>
</Modal>
`;

exports[`renders I3BottomSheet correctly should render I3BottomSheet with useSwipeDown={true} and title="Modal Title" 1`] = `
<Modal
  animationIn="slideInUp"
  animationOut="slideOutDown"
  hasBackdrop={true}
  isVisible={true}
  onBackdropPress={[MockFunction]}
  onSwipeComplete={[MockFunction]}
  style={
    {
      "margin": 0,
    }
  }
  swipeDirection="down"
  useNativeDriverForBackdrop={true}
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
          "marginTop": "auto",
          "padding": 16,
          "paddingBottom": 40,
        },
        {
          "paddingTop": 8,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#000",
          "borderRadius": 3,
          "height": 5,
          "width": 45,
        }
      }
    />
    <View
      style={
        {
          "height": 32,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "height": 16,
            "width": undefined,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 16,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        Modal Title
      </Text>
    </View>
    <View
      style={
        {
          "height": 16,
          "width": undefined,
        }
      }
    />
    <Text>
      should render I3BottomSheet with useSwipeDown=
    </Text>
  </View>
</Modal>
`;
